### PR TITLE
add teams app staging

### PR DESCRIPTION
**Pull request by laujan**

### CODE REVIEW

File: api-reference/v1.0/resources/teamsappdefinition.md [hunk 1]
```diff
@@ -21,6 +19,8 @@ The details of one version of a [teamsApp](teamsapp.md).
 |:------------------- |:-------- |:----------- |
 | id                  | string   | A unique id (not the teams appid). |
 | teamsAppId          | string   | The id from the Teams App manifest. |
+| publishingState| string|The published status of a specific version of a Teams app. Values are `published` or `rejected`.|
```
Comment (subray2014) [inline api-reference/v1.0/resources/teamsappdefinition.md:22]: There are 3 states:
1. submitted - The specific version of the Teams App has been submitted and is under review.
2. rejected - The request to publish the specific version of the Teams App was rejected by the admin.
3. published - The request to publish the specific version of the Teams App has been approved by the admin and the app is published now.

Please refer:
[https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=/reviews/4574-teams-apps-staging-apis/api.md&_a=preview&version=GBmaster](https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=/reviews/4574-teams-apps-staging-apis/api.md&_a=preview&version=GBmaster)

File: api-reference/v1.0/resources/teamsappdefinition.md [hunk 2]
```diff
@@ -21,6 +19,8 @@ The details of one version of a [teamsApp](teamsapp.md).
...
+|requiresReview|boolean|This triggers the review process if set to `true`. If not set, or set to `false`, the app is considered approved and instantly published.|
```
Comment (subray2014) [inline api-reference/v1.0/resources/teamsappdefinition.md:23]: RequiresReview is not a property of the teamsAppDefinition. It's a query parameter to the Create and Update APIs.

File: api-reference/v1.0/api/teamsapp-publish.md [hunk 3]
```diff
@@ -26,6 +24,7 @@ One of the following permissions is required to call this API. To learn more, in
 | Permission Type                        | Permissions (from least to most privileged)|
 |:----------------------------------     |:-------------|
 | Delegated (work or school account)     | AppCatalog.ReadWrite.All, Directory.ReadWrite.All |
+| Delegated (work or school account) | App.Catalog.Submit |
```
Comment (subray2014) [inline api-reference/v1.0/api/teamsapp-publish.md:27]: The note above, ">**Note:** Only global administrators can call this API.", needs to be changed.
With the new permission "AppCatalog.Submit", non-admin users can also submit apps for review.
They must use the requiresReview=true query parameter during submissions.

File: api-reference/v1.0/api/teamsapp-publish.md [hunk 4]
```diff
@@ -35,6 +34,13 @@ One of the following permissions is required to call this API. To learn more, in
 POST /appCatalogs/teamsApps
 ```
 
+```http
```
Comment (subray2014) [inline api-reference/v1.0/api/teamsapp-publish.md:37]: These should be in different sections. Examples are also needed.
Please refer:
[https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=/reviews/4574-teams-apps-staging-apis/api.md&_a=preview&version=GBmaster](https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=/reviews/4574-teams-apps-staging-apis/api.md&_a=preview&version=GBmaster)

File: api-reference/v1.0/api/teamsapp-delete.md [hunk 5]
```diff
@@ -48,9 +52,9 @@ None.
 
 ## Response
 
-```
```
Comment (nkramer) [inline api-reference/v1.0/api/teamsapp-delete.md:51]: Don't we want triple ```?

File: api-reference/v1.0/api/teamsapp-list.md [hunk 6]
```diff
@@ -115,3 +115,59 @@ Content-Type: application/json
   ]
 }
 ```
+
+### Example 3: List applications with the associated app definitions and progress in the staging approval pipeline
```
Comment (subray2014) [inline api-reference/v1.0/api/teamsapp-list.md:119]: Is "staging" the word we use? Should it instead be review pipeline? I would confirm with Nick.

File: api-reference/v1.0/api/teamsapp-update.md [hunk 7]
```diff
@@ -51,26 +49,90 @@ Do not use the ID from the manifest of the zip app package.
...
+         "Channel.Create.Group",
+         "Tab.ReadWrite.Group",
+         "Member.Read.Group"
+    ],
+    "publishingState": "submitted",
+    "lastModifiedDateTime": "2020-02-10 22:48:33.841",
+}
+```
+
+### Example 3: Admin approval of an application pending review
```
Comment (subray2014) [inline api-reference/v1.0/api/teamsapp-update.md:116]: Please remove example 3. Admin approval feature is not yet complete.

**Pull request closed without merging**